modules/6_clusterqc/rna/clusterqc.qmd

```diff
@@ -0,0 +1,309 @@
+
+
+# Load libraries
+library(knitr)
+library(magrittr)
+library(gt)
+library(Seurat)
+library(ggplot2)
+library(future)
+library(patchwork)
+
+# Get module name and directory (needed to access files within the module directory)
+module_name = params$module_name
+module_dir = params$module_dir
+
+# Parallelisation plan for all functions that support future
+plan(multisession, workers=4, gc=TRUE)
+
+# Be verbose
+verbose=FALSE
+```
+
+```{r}
+#| label: clusterqc_rna_preparation
+
+###############
+# Directories #
+###############
+
+# Module directory 'results' contains all output that should be provided together with the final report of the project
+dir.create(file.path(module_dir, "results"), showWarnings=FALSE)
+files = list.files(path=file.path(module_dir, "results"), full.names=TRUE)
+if (length(files) > 0) unlink(files, recursive=TRUE)
+
+# Module directory 'sc' contains the final Seurat object
+dir.create(file.path(module_dir, "sc"), showWarnings=FALSE)
+files = list.files(path=file.path(module_dir, "sc"), full.names=TRUE)
+if (length(files) > 0) unlink(files, recursive=TRUE)
+
+#################
+# Seurat object #
+#################
+
+# Read in seurat object
+if (!is.null(param("prev_module_dir"))) {
+  prev_module_dir = param("prev_module_dir")
+} else {
+  prev_module_dir = PreviousModuleDir(module_dir)
+}
+prev_sc_obj = file.path(prev_module_dir, "sc", "sc.rds")
+if(!file.exists(prev_sc_obj)) {
+  stop(FormatMessage("Could not find a sc.rds file in {{prev_module_dir}}. Was the respective module already run?"))
+}
+sc = SeuratObject::LoadSeuratRds(prev_sc_obj)
+
+# Move on-disk layers to faster temp location if requested
+on_disk_counts = param("on_disk_counts")
+on_disk_use_tmp = param("on_disk_use_tmp")
+
+if (on_disk_counts & on_disk_use_tmp) {
+  on_disk_path = tempdir()
+  with_progress({
+    sc = UpdateMatrixDirs(sc, dir=on_disk_path)
+  }, enable=verbose)
+} else {
+  on_disk_path = file.path(module_dir, "sc")
+}
+
+# sc = SeuratObject::LoadSeuratRds("modules/3_normalization/rna/sc/sc.rds")
+
+# Set default assay
+default_assay = param("default_assay")
+Seurat::DefaultAssay(sc) = default_assay
+```
+
+```{r}
+#| label: temporary
+#| warning: false
+#| message: false
+sc <- suppressMessages(ScaleData(sc))
+sc <- suppressMessages(RunPCA(sc))
+sc <- suppressMessages(FindNeighbors(sc, dims = 1:10))
+sc <- suppressMessages(FindClusters(sc, resolution = 0.5))
+sc <- suppressMessages(RunUMAP(sc, dims = 1:10))
+```
+## Quality Control 
+
+Do cells in individual clusters have particularly high counts, detected genes or mitochondrial content?
+
+```{r}
+#| label: fig-clusterqc
+#| warning: false
+#| message: false
+#| results: asis
+
+qc_features = paste(c("nCount", "nFeature", "pMito", "pRibosomal", "pGlobin", "pERCC", "pXIST", "pChrY"), default_assay, sep="_")
+if ("DoubletScore" %in% colnames(sc[[]])) qc_features = c(qc_features, "DoubletScore")
+
+# Feature plots
+plist1 = Seurat::FeaturePlot(sc, features=qc_features, combine=FALSE)
```

Comment:
Can you please add `raster=ncol(sc)>=getOption("raster.threshold")` as argument? 

If the number of cells exceeds `raster.threshold` (defined in `general_configuration.R`) points will be rastered into an image in all pdfs so that you do not have thousands of objects.